src/github.com/getlantern/idletiming/idletiming_conn.go

```diff
@@ -91,6 +100,13 @@ func (c *IdleTimingConn) TimesOutAt() time.Time {
 
 // Read implements the method from io.Reader
 func (c *IdleTimingConn) Read(b []byte) (int, error) {
+	c.closeMutex.RLock()
+	defer c.closeMutex.RUnlock()
```

Comment:
The purpose of the lock is to prevent us from closing the underlying connection while an io op is in progress.  That requires us to hold the lock for the duration of the method.